### PR TITLE
refactor(elasticsearch-exporter): output raw message on JSON error

### LINKED ISSUE #10970: Poor error handling in Elasticsearch exporter of incorrect response parsing
**Describe the bug**

If you configure Elastic behind a reverse proxy, or in any way which would cause your request to be redirected, you may some time send a request in the exporter (whether exporting a bulk, or putting an index template) and receive HTML. 

> This occurred in a [support case](https://jira.camunda.com/browse/SUPPORT-15017).

If you see such an error when parsing a response, e.g.

```
Caused by: com.fasterxml.jackson.core.JsonParseException: Unexpected character ('<' (code 60)): expected a valid value (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
```

This is highly likely because the server returned HTML (or possible XML, but less likely).

In this case, as there was no request error, it means the server returned something between 200 and 399. The root cause is not yet clear.

**To Reproduce**

Still unsure, I will try to get this from the support case. However, I can say that it is connecting Zeebe to an external, self-managed Elasticsearch (so not deployed via the Helm charts). The version however should be compatible, 7.13.x.

**Expected behavior**

If it happens to be a redirect, we should handle redirects properly. If the response is truly invalid JSON, then we should print out the raw response in the exception so we can diagnose what's happening.

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>

```
2022-11-08 06:16:52.573 [Broker-0-Exporter-1] [Broker-0-zb-fs-workers-0] WARN 
      io.camunda.zeebe.broker.exporter.elasticsearch - Error on exporting record with key 2251799813685249
io.camunda.zeebe.exporter.ElasticsearchExporterException: Failed to put component template
	at io.camunda.zeebe.exporter.ElasticsearchClient.putComponentTemplate(ElasticsearchClient.java:215) ~[zeebe-elasticsearch-exporter-8.1.2.jar:8.1.2]
	at io.camunda.zeebe.exporter.ElasticsearchClient.putComponentTemplate(ElasticsearchClient.java:145) ~[zeebe-elasticsearch-exporter-8.1.2.jar:8.1.2]
	at io.camunda.zeebe.exporter.ElasticsearchExporter.createComponentTemplate(ElasticsearchExporter.java:209) ~[zeebe-elasticsearch-exporter-8.1.2.jar:8.1.2]
	at io.camunda.zeebe.exporter.ElasticsearchExporter.createIndexTemplates(ElasticsearchExporter.java:147) ~[zeebe-elasticsearch-exporter-8.1.2.jar:8.1.2]
	at io.camunda.zeebe.exporter.ElasticsearchExporter.export(ElasticsearchExporter.java:79) ~[zeebe-elasticsearch-exporter-8.1.2.jar:8.1.2]
	at io.camunda.zeebe.broker.exporter.stream.ExporterContainer.lambda$export$3(ExporterContainer.java:156) ~[zeebe-broker-8.1.2.jar:8.1.2]
	at io.camunda.zeebe.util.jar.ThreadContextUtil.runCheckedWithClassLoader(ThreadContextUtil.java:58) ~[zeebe-util-8.1.2.jar:8.1.2]
	at io.camunda.zeebe.util.jar.ThreadContextUtil.runWithClassLoader(ThreadContextUtil.java:34) ~[zeebe-util-8.1.2.jar:8.1.2]
	at io.camunda.zeebe.broker.exporter.stream.ExporterContainer.export(ExporterContainer.java:155) ~[zeebe-broker-8.1.2.jar:8.1.2]
	at io.camunda.zeebe.broker.exporter.stream.ExporterContainer.exportRecord(ExporterContainer.java:142) ~[zeebe-broker-8.1.2.jar:8.1.2]
	at io.camunda.zeebe.broker.exporter.stream.ExporterDirector$RecordExporter.export(ExporterDirector.java:523) ~[zeebe-broker-8.1.2.jar:8.1.2]
	at io.camunda.zeebe.scheduler.retry.BackOffRetryStrategy.run(BackOffRetryStrategy.java:51) ~[zeebe-scheduler-8.1.2.jar:8.1.2]
	at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:94) ~[zeebe-scheduler-8.1.2.jar:8.1.2]
	at io.camunda.zeebe.scheduler.ActorJob.execute(ActorJob.java:45) ~[zeebe-scheduler-8.1.2.jar:8.1.2]
	at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:119) ~[zeebe-scheduler-8.1.2.jar:8.1.2]
	at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:106) ~[zeebe-scheduler-8.1.2.jar:8.1.2]
	at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:87) ~[zeebe-scheduler-8.1.2.jar:8.1.2]
	at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:198) ~[zeebe-scheduler-8.1.2.jar:8.1.2]
Caused by: com.fasterxml.jackson.core.JsonParseException: Unexpected character ('<' (code 60)): expected a valid value (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: (org.apache.http.nio.entity.ContentInputStream); line: 1, column: 2]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:2391) ~[jackson-core-2.13.4.jar:2.13.4]
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:735) ~[jackson-core-2.13.4.jar:2.13.4]
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportUnexpectedChar(ParserMinimalBase.java:659) ~[jackson-core-2.13.4.jar:2.13.4]
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._handleUnexpectedValue(UTF8StreamJsonParser.java:2737) ~[jackson-core-2.13.4.jar:2.13.4]
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._nextTokenNotInObject(UTF8StreamJsonParser.java:902) ~[jackson-core-2.13.4.jar:2.13.4]
```

</p>
</details>

**Environment:**
- OS: Ubuntu Focal (from image)
- Zeebe Version: 8.1.2
- Configuration: self-managed, external Elasticsearch (self deployed)

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

Ensures that the raw message is printed as part of the stacktrace when the server returns a successful, but invalid message, so we can more easily diagnose what is happening.

## Related issues

closes #10970 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: exporters/elasticsearch-exporter/src/main/java/io/camunda/zeebe/exporter/ElasticsearchClient.java [hunk 1]
```diff
@@ -162,7 +162,8 @@ private void exportBulk() {
 
     final BulkIndexResponse response;
     try {
-      response = MAPPER.readValue(httpResponse.getEntity().getContent(), BulkIndexResponse.class);
+      final var responseBody = httpResponse.getEntity().getContent();
+      response = MAPPER.readValue(responseBody.readAllBytes(), BulkIndexResponse.class);
```
Comment (Zelldon) [inline exporters/elasticsearch-exporter/src/main/java/io/camunda/zeebe/exporter/ElasticsearchClient.java:166]: :thinking: I'm wondering why we read the response in all theses cases even if the response might be failed (4x, 5x)?

**Review by ChrisKujawa — APPROVED** [review]
I guess I'm a bit rusty in the HTTP client API, but I got it now after a while :D Thanks :+1:

File: exporters/elasticsearch-exporter/src/main/java/io/camunda/zeebe/exporter/ElasticsearchClient.java (hunk 1 above)
Comment (npepinpe) [reply exporters/elasticsearch-exporter/src/main/java/io/camunda/zeebe/exporter/ElasticsearchClient.java:166]: The client will throw an exception on 4xx or 5xx, so there's no need to worry about it here. What happened in the support case is the response was not an error, but it was a proxy returning some content.

**Comment by npepinpe** [comment]
Your comment made me realize I should make it more explicit why we're doing this. Someone who's unaware of this error reporting behavior but aware that Jackson can deserialize an `InputStream` may, legitimately, revert this for a low hanging fruit, performance-wise.

**Comment by npepinpe** [comment]
bors merge

**Pull request merged**